robottelo/hosts.py

```diff
@@ -561,10 +561,14 @@ def create_custom_repos(self, **kwargs):
             baseurl=http://repourl.domain.com/path
             enabled=1
             gpgcheck=0
+            proxy=http://proxy.domain.com:1234
 
         """
         for name, url in kwargs.items():
-            content = f'[{name}]\n' f'name={name}\n' f'baseurl={url}\n' 'enabled=1\n' 'gpgcheck=0'
+            content = f'[{name}]\n' f'name={name}\n' f'baseurl={url}\n' 'enabled=1\n' 'gpgcheck=0\n'
+            # Add proxy configuration for IPv6
+            if settings.server.is_ipv6:
+                content += f'proxy={settings.http_proxy.http_proxy_ipv6_url}'
```

Comment:
To me this looks rather magical. I don't know robottello well enough, but isn't there a better way?

Is there a tracker to add IPv6 support to the host that's missing? Would it be better to set up a reverse proxy just for the single host that's missing or instead use RHEL from satellite.sat which is IPv6 enabled?

Looking at the various tests this code is also used in other places where it could have negative side effects.

Reply:
I'm not aware if there's any tracker to add IPv6 support to the host, maybe @jyejare knows if there's any ?

Reply:
@ekohl @shubhamsg199 Some Comments and questions:

- `Is there a tracker to add IPv6 support to the missing host?`
Do you mean a tracker in Jira or in robottelo automation? 
   BTW, the proxy here must be used in case of IPv6 host is not able to reach to the external source for contents.  The tracking for ipv6 is auto-handled as the test is modified to spin up the ipv6 host when the ipv6 satellite testing is running. 

- `Would it be better to set up a reverse proxy just for the single host that's missing`
Whats the purpose here to setup a reverse proxy ?

- `Instead use RHEL from satellite.sat which is IPv6 enabled?`
The contents for RHEL are not different in IPv6 and Ipv4 so does not matter from where you get them. I may misunderstood your idea, please feel free to elaborate.


- `Looking at the various tests this code is also used in other places where it could have negative side effects.`
Yes, that's right the impact is there when the proxy would still be used even when the Ipv4 host is spinned up. So determining the host network would be a crucial but not critical. Especially this would be the case for cross network testing and atleast for now we are not there yet.

Reply:
@jyejare IIUC I think for tracker @ekohl means if we've some JIRA/SNOW ticket to make `download.devel..` host dual stack
And, I think it will take time since its not managed by our team, so until that's made dual stack we can configure proxy for IPv6 runs only, and it won't configure proxy in case of IPv4 run, so I feel it won't have any side effects on IPv4 results and It is only used for content host testing, not for satellite installations

Reply:
@ekohl Ohh Dual Stack. The dual stack is coming soon and we will parametrizing the content hosts to run on ipv4, ipv6 and dual stack sooner. And dual stack should not need proxy.

Hence I said above the proxy enablement should happen only when ipv6-only content host test is running.